text_generation/causal_lm/cpp/greedy_causal_lm.cpp

```diff
@@ -91,8 +91,16 @@ int main(int argc, char* argv[]) try {
     lm.get_tensor("input_ids").set_shape({BATCH_SIZE, 1});
     position_ids.set_shape({BATCH_SIZE, 1});
     TextStreamer text_streamer{std::move(detokenizer)};
-    // There's no way to extract special token values from the detokenizer for now
-    constexpr int64_t SPECIAL_EOS_TOKEN = 2;
+    // After compiling the model, before the inference loop
+    auto rt_info = lm_model.get_rt_info();
```

Comment:
I suppose `eos_token_id` is a property of tokenizer model, but LLM one

Reply:
I'm sorry but what do you mean by the LLM one?

Reply:
I have made the changes. Sorry about that!

Reply:
it's ok, you don't have to apologize :) 
In order to make it work and to merge there are some other corrections. Let's fix them. I will close this as resolved and write it PR discussion.